clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -447,10 +552,36 @@ public ConfigDef withClientSaslSupport() {
             String joined = Utils.join(undefinedConfigKeys, ",");
             throw new ConfigException("Some configurations in are referred in the dependents, but not defined: " + joined);
         }
-        // parse all known keys
+        // Handle all known ConfigKey objects.
         Map<String, Object> values = new HashMap<>();
-        for (ConfigKey key : configKeys.values())
-            values.put(key.name, parseValue(key, props.get(key.name), props.containsKey(key.name)));
```

Comment:
We no longer use `parseValue` here, but `valuesWithPrefixOverride` still uses it. Will the latter handle deprecated key names correctly?

Reply:
That's a good point.  I fixed `valuesWithPrefixOverride` to work with deprecated config keys and added a unit test.  I also removed `parseValue`.